nexus/db-queries/src/db/queries/network_interface.rs

```diff
@@ -1502,18 +1504,39 @@ fn push_instance_state_verification_subquery<'a>(
 ///                     parent_id = <parent_id> AND
 ///                     kind = <kind> AND
 ///                     time_deleted IS NULL
-///             ) = 1,
+///             ) <= 1,
```

Comment:
This `= 1` is where we were triggering the incorrect "can't delete while secondaries exist" error. Specifically, one of two things has to be true in this check; either:

a) this interface is a secondary interface that is not deleted (lines 1489-1495), or
b) there is exactly 1 interface for this instance/service that is not deleted (lines 1499-1507)

If we're only considering instances/services with a single, primary interface, (a) is never true, so we always check (b). If we've already deleted the only interface, this count returns 0, not one, triggering the `'secondaries'` error sentinel. The change here is to allow "at most 1" instead of "exactly 1", so we get past this check and proceed to the rest of the CTE if there are no undeleted interfaces left.